### PR TITLE
Overloaded 'contains' operator for string regex matching

### PR DESCRIPTION
**Pull request by VatoKo**
## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 5.0.
- [x] New extensions support iOS 10.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+, or use `@available` if not.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

File: Sources/SwifterSwift/SwiftStdlib/StringExtensions.swift [hunk 1]
```diff
@@ -957,6 +957,17 @@ public extension String {
         return range(of: pattern, options: .regularExpression, range: nil, locale: nil) != nil
     }
     #endif
+    
+    #if canImport(Foundation)
+    /// SwifterSwift: Overload Swift's 'contains' operator for matching regex pattern
+    ///
+    /// - Parameter lhs: String to check on regex pattern.
+    /// - Parameter rhs: Pattern to verify.
```
Comment (LucianoPAlmeida) [inline Sources/SwifterSwift/SwiftStdlib/StringExtensions.swift:965]:
```suggestion
    /// - Parameter rhs: Regex pattern to match against.
```

File: Tests/SwiftStdlibTests/StringExtensionsTests.swift [hunk 2]
```diff
@@ -576,6 +576,16 @@ final class StringExtensionsTests: XCTestCase {
...
         XCTAssertTrue("email@mail.com".matches(pattern: "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}"))
     }
+    
+    func testRegexMatchOperator() {
+        #if canImport(Foundation)
+        XCTAssertTrue("123" ~= "\\d{3}")
+        XCTAssertFalse("dasda" ~= "\\d{3}")
+        XCTAssertFalse("notanemail.com" ~= "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}")
+        XCTAssertTrue("email@mail.com" ~= "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}")
+        XCTAssertTrue("hat" ~= "[a-z]at")
```
Comment (LucianoPAlmeida) [inline Tests/SwiftStdlibTests/StringExtensionsTests.swift:586]: Could you add empty strings test cases?

Comment (VatoKo) [reply Tests/SwiftStdlibTests/StringExtensionsTests.swift:586]: Yeah, sure

File: Sources/SwifterSwift/SwiftStdlib/StringExtensions.swift [hunk 3]
```diff
@@ -957,6 +957,17 @@ public extension String {
...
     #endif
+    
+    #if canImport(Foundation)
+    /// SwifterSwift: Overload Swift's 'contains' operator for matching regex pattern
+    ///
+    /// - Parameter lhs: String to check on regex pattern.
+    /// - Parameter rhs: Regex pattern to match against.
+    /// - Returns: true if string matches the pattern.
+    static func ~= (lhs: String, rhs: String) -> Bool {
+        return lhs.range(of: rhs, options: .regularExpression, range: nil, locale: nil) != nil
```
Comment (guykogus) [inline Sources/SwifterSwift/SwiftStdlib/StringExtensions.swift:968]: You can use the default values.
```suggestion
        return lhs.range(of: rhs, options: .regularExpression) != nil
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,9 @@ The changelog for **SwifterSwift**. Also see the [releases](https://github.com/S
 ## Upcoming Release
 
 ### Added
+
+-**String**:
+  - Overloaded Swift's 'contains' operator for string to check regex mathing. [#858](https://github.com/SwifterSwift/SwifterSwift/pull/858) by [VatoKo](https://github.com/VatoKo)
```
Comment (guykogus) [inline CHANGELOG.md:10]:
```suggestion
  - Overloaded Swift's 'contains' operator (`~=`) for `String` to check regex matching. [#858](https://github.com/SwifterSwift/SwifterSwift/pull/858) by [VatoKo](https://github.com/VatoKo)
```

File: Tests/SwiftStdlibTests/StringExtensionsTests.swift [hunk 5]
```diff
@@ -576,6 +576,19 @@ final class StringExtensionsTests: XCTestCase {
         XCTAssertFalse("notanemail.com".matches(pattern: "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}"))
         XCTAssertTrue("email@mail.com".matches(pattern: "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}"))
     }
+    
+    func testRegexMatchOperator() {
+        #if canImport(Foundation)
```
Comment (guykogus) [inline Tests/SwiftStdlibTests/StringExtensionsTests.swift:581]: Sorry to be a pain, but can you move the `#if` to outside the function?

Comment (VatoKo) [reply Tests/SwiftStdlibTests/StringExtensionsTests.swift:581]: Sure, no problem

**Comment by SuperY** [comment]
Additional test case:
```
    func testPatternMatchOperator() {
        let codeString = "0"
        switch codeString {
        case "101":
            XCTAssert(codeString == "101") //  Not pass here.
        case "0":
            XCTAssert(codeString == "0")
        default:
            XCTExpectFailure("Switch string value, not matching the correct result.")
        }
   }
```

**Pull request merged**